### PR TITLE
Add "skip_deletion" option for vkcs_db_user

### PR DESCRIPTION
**Pull request by paaanic**
Allow user to skip deletion of Databases user, this is useful when deleting PostgreSQL users that own some database objects and therefore cannot be dropped.

CMPT-33820

### CODE REVIEW

File: vkcs/db/resource_user.go [hunk 1]
```diff
@@ -77,6 +77,23 @@ func ResourceDatabaseUser() *schema.Resource {
...
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"skip_deletion": {
+							Type:        schema.TypeBool,
+							Default:     false,
+							Optional:    true,
+							Description: "Boolean to control whether to user deletion should be skipped. If set to true, the resource will be removed from sthe tate, but the remote object will not be deleted. This is useful for PostgreSQL, where users cannot be deleted from the API if they own database objects.",
```
Comment (schirevko) [inline vkcs/db/resource_user.go:92]: typo in "from sthe tate"

**Review by schirevko — CHANGES_REQUESTED** [review]

File: vkcs/db/resource_user.go [hunk 2]
```diff
@@ -304,6 +321,20 @@ func resourceDatabaseUserUpdate(ctx context.Context, d *schema.ResourceData, met
 
 func resourceDatabaseUserDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(clients.Config)
+	var skipDeletion bool
+
+	vendorOptionsRaw := d.Get("vendor_options").(*schema.Set)
+	if vendorOptionsRaw.Len() > 0 {
+		vendorOptions := util.ExpandVendorOptions(vendorOptionsRaw.List())
+		if v, ok := vendorOptions["skip_deletion"]; ok || v.(bool) {
+			skipDeletion = true
```
Comment (ftersin) [inline vkcs/db/resource_user.go:330]: why do not exit here?

File: vkcs/db/resource_user.go (hunk 1 above)
Comment (paaanic) [reply vkcs/db/resource_user.go]: Fixed.

File: vkcs/db/resource_user.go (hunk 2 above)
Comment (paaanic) [reply vkcs/db/resource_user.go]: Updated.

**Review by ftersin — APPROVED** [review]

**Review by schirevko — APPROVED** [review]

**Pull request merged**